EIPS/eip-9999.md

```diff
@@ -0,0 +1,447 @@
+
+    if execution_result.error:
+        return ChunkExecutionResult(
+            status="INVALID",
+            chunk_index=chunk_index,
+            error=execution_result.error,
+        )
+    
+    return ChunkExecutionResult(
+        status="VALID",
+        chunk_index=chunk_index,
+    )
+```
+
+
+### Network Protocol
+
+#### Chunk Propagation
+
+Chunks and CALs are identified by beacon block root and chunk index. They propagate on separate subnets.
+
+- **Chunks**: Published on `execution_chunk_sidecar` subnet
+- **CALs**: Published on `chunk_access_list_sidecar` subnet
+
+Nodes MUST subscribe to both subnets to ensure full block availability. Chunks and CALs received before the beacon block are stored as "orphans" and validated when the matching block arrives.
+
+### Consensus Layer Changes
+
+The consensus layer orchestrates chunk execution as beacon block, chunks and CALs become available.
+
+#### Store Extensions
+
+```python
+class Store:
+    # ... existing fields ...
+
+    # Tracks whether CALs were received
+    chunk_access_lists: Dict[Root, Set[ChunkIndex]]
+    # Tracks chunk headers 
+    chunk_headers: Dict[Root, Dict[ChunkIndex, ExecutionChunkHeader]]
+    # Tracks chunk's execution status
+    chunk_execution_statuses: Dict[Root, Dict[ChunkIndex, ExecutionStatus]]
+    # Tracks block validation
+    block_valid: Dict[Root, bool]
+```
+
+#### Three-Phase Validation
+
+##### Phase 1: Forward CALs to EL client
+
+Since CALs are required for validating all subsequent chunks, we forward them to the EL client as soon as they arrive. This avoids sending the same CAL multiple times between CL and EL.
+
+```python
+def on_chunk_access_list_received(store, cal_sidecar):
+    root = cal_sidecar.root
+    chunk_index = cal_sidecar.chunk_index
+
+    # Get or wait for beacon block
+    block = wait_beacon_block(store, root)
+    # Verify sidecar
+    assert verify_chunk_access_list_sidecar(block, cal_sidecar)
+
+    # Send CAL to EL client
+    execution_engine.new_chunk_access_list(block, chunk_index, cal_sidecar.chunk_access_list)
+
+    # Track that CAL was received and unblock any ongoing validation
+    store.chunk_access_lists[root].add(chunk_index)
+    notify_cal_available(root, chunk_index) # unblock wait_cal function
+```
+
+##### Phase 2: Streaming Chunk Validation
+
+Chunks are validated as soon as they and all required CALs are available. 
+
+```python
+def on_chunk_received(store, chunk_sidecar):
+    root = chunk_sidecar.root
+    chunk = chunk_sidecar.chunk
+    chunk_header = chunk.chunk_header
+
+    # Get or wait for beacon block
+    block = wait_beacon_block(store, root)
+    # Verify sidecar
+    assert verify_chunk_sidecar(block, chunk_sidecar)
+
+    # Store chunk header
+    store.chunk_headers[root][chunk_header.index] = chunk_header
+
+    # Wait prior and chunk's CALs
+    for chunk_index in range(chunk_header.index + 1):
+        wait_cal(root, chunk_index)
```

Comment:
again i think instead of an explicit wait, we could queue up the chunks but i think that could also be an impl detail.